### ISSUE TITLE
Show Search Results in Notes Field Context

### ISSUE THREAD

**Issue by kentborg** [keepassxreboot/keepassxc#11005]
## Summary

I need to find things within a notes field that is long.

For some entries I have long text in the "Notes" field. I like that searching will find things **in** the notes field. but I don't know how to see **where** in the notes field.

Sometimes I copy and paste the notes field into a text editor just so I can search…but that's a scary approach.

## Context

Why is my notes field so long? Usually because for my current job I have lots of stuff that I want save and free from text is an easy way. What is my employee number? My hire date? Other HR crap. Maybe a phone number or two. My made up random answers to stupid security questions. Some damn Jetbrains login that has gone away since my layoff but was important.

Or notes about my ISP, this isn't long like for a job, but long enough to scroll: Account number, circuit number, MAC addresses, router password, notes about trouble tickets, security questions again, etc.

My credit card: The account number, expiration date, CCV, that funny number next to the signature that will rub off by the time they ask me for it to verify I have the card in my possession, security questions, notes about disputed charges, etc.

I don't want to have to structure this into separate entries when the data is so heavily grouped.

## Possible Design Options

- Remember what the last search was, and when an entry is open have search again highlight each hit in turn (scrolling into view if necessary).
- Implement a new search feature: When I have an entry open and allow me to search, scroll the hit into view and highlight it, allow search again to scroll and highlight the next hit. 

Maybe these two option are effectively the same, I'm not sure.

Probably also a good idea to add an Edit menu: a place to put cut, copy, paste, delete, undo, redo, **search**, and **search again**. Menus are a compact way to expose features and document their keyboard equivalents.

## Generally, Thanks!

I appreciate your work. I also really like your refusing to add features that would be dangerous. I don't use an online password manager or service—which strives for seamless browser integration!—because those features are dangerous.

I'll backup my encrypted database to the cloud myself, but only after I superencrypt it with a better passphrase than the (somewhat) typeable one I use day-to-day.

And I finally looked into how crackable the database format is and am pleased to discover that all the cracking seems to depend on my choosing a weak passphrase. :whew:

**Comment by droidmonkey** [comment]
Duplicate of #4864, thank you for the kind words! Might I recommend using advanced attributes for the more specific fields you listed such as employee number and such. You can then use various features like copy to clipboard and auto-type to put that information into forms or other places.

**Comment by kentborg** [comment]
I'll play with advanced attributes, see whether it helps.

I will point out that I didn't find it at all obvious what "advanced attributes" means, and it is a surprise to me that it includes the ability add…what seem to be new hidden-ish fields. (Hidden-ish: I don't see them when looking at the regular fields.)

Usually I am pretty willing to play with such features, to see what they do and how they work. But the data I have in keepass is _**very**_ important, and that makes me reluctant to mess with things that look like they make persistent changes, or maybe will just confuse me and lead to user error.

In #4864 you wrote: "Adding this feature would not be impossible, but add complexity that almost no one will use."

I disagree. There have to be a lot of people with long notes fields. I expect they would all be delighted if they could search within that text.

I have long notes fields for entries other than employers: also banks, brokerage houses, the post office, ISPs, my cellphone provider, VoIP provider, TV streaming service, domain registrar, credit cards…and this is not extraneous data, it is all directly dealing with those accounts; some have dated annotations about my dealings with them, and some of this stuff is quite sensitive: More than one of of my accounts has more than one PIN, and when I am on the phone they ask me for the telephone PIN, I have to scroll around looking for it. 

Being able to type new things directly when talking on the phone, and see old things (maybe what answers did I give to the security questions), without having to click around in the UI, would be quite valuable.

As others over on #4864 mention, sometimes I will also resort to pasting the notes field into a bare-bones text editor where the only missing feature I am looking for is search.

Please think about it…

Thanks,

-kb

P.S. Of course, please listen to me on _this_, but remain resolutely stubborn about never adding plugins, no matter how much others might _beg!_ Because plugins are both complex (bad for security) and seemingly designed to be dangerous—as users start collecting both sloppy and intentionally evil plugins.

(You do not have an easy job here, I know that.)
